### PR TITLE
[Gradle] - Turn off java doc lint, turn java and groovy warnings to errors

### PR DESCRIPTION
**Pull request by jaredsburrows**
@benjamin-bader 
- Turn off annoying Javadoc lint
- Turn Java warnings into errors
- Turn Groovy warnings into errors

### CODE REVIEW

File: src/main/groovy/com/getkeepsafe/dexcount/DexCountException.groovy [hunk 1]
```diff
@@ -17,6 +17,8 @@
 package com.getkeepsafe.dexcount
 
 final class DexCountException extends RuntimeException {
```
Comment (jaredsburrows) [inline src/main/groovy/com/getkeepsafe/dexcount/DexCountException.groovy]: warning: [serial] serializable class DexCountException has no definition of serialVersionUID
public class DexDataException extends RuntimeException {
       ^
error: warnings found and -Werror specified
1 error
1 warning

File: src/main/java/com/android/dexdeps/DexDataException.java [hunk 2]
```diff
@@ -20,4 +20,5 @@
  * Bad data found inside a DEX file.
  */
 public class DexDataException extends RuntimeException {
+    private static final long serialVersionUID = 1L;
```
Comment (jaredsburrows) [inline src/main/java/com/android/dexdeps/DexDataException.java]:
```
warning: [serial] serializable class DexDataException has no definition of serialVersionUID
public class DexDataException extends RuntimeException {
       ^
error: warnings found and -Werror specified
1 error
1 warning
```

Comment (benjamin-bader) [inline src/main/java/com/android/dexdeps/DexDataException.java]: Possible to disable warnings for the Java sources?  They're all sourced from AOSP, and I try to touch them as little as possible (with one notable exception).

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# EditorConfig helps developers define and maintain consistent
+# coding styles between different editors and IDEs
+# editorconfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+# Change these settings to your own preference
```
Comment (jaredsburrows) [inline .editorconfig]: @benjamin-bader  To fix formatting for everyone.

**Comment by jaredsburrows** [comment]
@benjamin-bader This PR is fixed.

**Pull request merged**